### PR TITLE
Remove is_gamma_argument() function

### PR DESCRIPTION
**Pull request by phate**
This PR does the following:
1. Introduces the GammaArgument class, which represents a region argument in a gamma subregion.
2. Removes the is_gamma_argument() function.

### CODE REVIEW

File: jlm/rvsdg/gamma.hpp [hunk 1]
```diff
@@ -465,6 +465,33 @@ inline gamma_node::gamma_node(jlm::rvsdg::output * predicate, size_t nalternativ
...
+private:
+  GammaArgument(
+      rvsdg::region & region,
+      structural_input & input,
+      std::shared_ptr<const rvsdg::type> type)
+      : argument(&region, &input, std::move(type))
+  {}
+
+  static GammaArgument &
+  Create(rvsdg::region & region, structural_input & input, std::shared_ptr<const rvsdg::type> type)
```
Comment (haved) [inline jlm/rvsdg/gamma.hpp:487]: Does the create method need a type specified, or can it get its type from the `input`?

**Review by haved — APPROVED** [review]
Looks good, just one question

File: jlm/rvsdg/gamma.hpp (hunk 1 above)
Comment (phate) [reply jlm/rvsdg/gamma.hpp]: Remove the type argument.

**Comment by phate** [comment]
@haved Changes since last review:
1. Removed superfluous type from `Create()` method and `GammaArgument` constructor
2. Handed in a `GammaInput` instead of a `structural_input` to `GammaArgument`'s constructor and `Create()` method

**Review by haved — APPROVED** [review]

**Pull request merged**